sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -2197,66 +2197,11 @@ private static void validateTimerFamilyField(
     return DoFnSignature.GetSizeMethod.create(m, windowT, methodContext.getExtraParameters());
   }
 
-  private static Collection<Method> declaredMethodsWithAnnotation(
```

Comment:
Overall idea looks good to me. I will leave @kennknowles  to comment this part as @kennknowles knows more about Java SDK.